src/ty/mod.rs

```diff
@@ -118,6 +118,16 @@ impl From<TypeDefTuple> for Type {
     }
 }
 
+impl From<TypeDefPhantom> for Type {
+    fn from(phantom: TypeDefPhantom) -> Self {
+        Self::new(
+            Path::prelude("PhantomData"),
+            vec![phantom.type_param],
```

Comment:
This is also arguably redundant since the type is already part of the definition, but it does make it consistent with e.g. `TypeDefArray` and `TypeDefTuple`

Reply:
No it's not consistent: all the others are `Self::new(Path::voldemort(), Vec::new(), $THING)`.

I did it this way to get what seemed like richer metadata when serializing, i.e. this:

```json
        "path": ["PhantomData"],
        "def": { "phantom": { "type": 1 } },
        "params": [1]
```

```json
        "def": { "phantom": { "type": 1 } },
```

…so I'll change this as suggested.